devicetypes/smartthings/zwave-motion-light-sensor.src/zwave-motion-light-sensor.groovy

```diff
@@ -0,0 +1,198 @@
+			break
+		case "0258": deviceWakeIntervalValue = 12  //Coolcam Neo reports once in 12h
+			break
+		default: deviceWakeIntervalValue = 4 //Default Z-Wave battery device reports once in 4h
+	}
+	return deviceWakeIntervalValue
+}
+
+private getCommandClassVersions() {
+	[
+			0x71: 3,  // Notification
+			0x20: 1,  // Basic
+			0x80: 1,  // Battery
+			0x72: 2,  // ManufacturerSpecific
+			0x31: 5,  // SensorMultilevel
+			0x84: 2,  // WakeUp
+			0x30: 2   //Sensor Binary
+	]
+}
+
+def parse(String description) {
+	def results = []
+	if (description.startsWith("Err")) {
+		results << createEvent(descriptionText: description, displayed: true)
+	} else {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			results += zwaveEvent(cmd)
+		}
+	}
+	log.debug "'$description' parsed to ${results.inspect()}"
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	return sensorMotionEvent(cmd.sensorValue)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	return sensorMotionEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def results = []
+	if (cmd.notificationType == 0x07) {          //Burglar
+		if (cmd.event == 0x08) {                 //detected
+			results << sensorMotionEvent(1)
+		} else if (cmd.event == 0x00) {          //inactive
+			results << sensorMotionEvent(0)
+		}
+	}
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def results = []
+	def map = [name: "battery", unit: "%", isStateChange: true]
+	state.lastbatt = now()
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	results << createEvent(map)
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def results = []
+	def map = [:]
+	switch (cmd.sensorType) {
+		case 3:
+			map.name = "illuminance"
+			map.value = cmd.scaledSensorValue.toInteger().toString()
+			map.unit = "lux"
+			map.isStateChange = true
+			break;
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	results << createEvent(map)
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def results = []
+	results << createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+	results << response(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 3, scale: 1).format())
+	if (!state.lastbatt || (now() - state.lastbatt) >= 10 * 60 * 60 * 1000) {
+		results << response(["delay 1000",
+							 zwave.batteryV1.batteryGet().format(),
+							 "delay 2000",
+							 zwave.wakeUpV2.wakeUpNoMoreInformation().format()
+		])
+	} else {
+		results << response(zwave.wakeUpV2.wakeUpNoMoreInformation().format())
```

Comment:
You could remove some code duplication here and just push this to the list after the `if` block.